### PR TITLE
Add form validation for signup and login routes

### LINKED ISSUE #17: Create API endpoint to handle POST request for chef sign up
/api/chefs/signup

### LINKED ISSUE #21: Create API endpoint to handle POST request for user sign up
/api/users/signup

### LINKED ISSUE #23: Create api endpoint to handle POST request for user login
/api/users/login

### LINKED ISSUE #25: Create API endpoint to handle POST request for chef login
/api/chefs/login

### LINKED ISSUE #58: Add authentication using JWT
Add login/signup auth scheme for users and chefs.

### PR DESCRIPTION
**Pull request by gzeegz**
Add some additional things for authentication like 
- token expiration
- additional information in payload
- form validation for signup and login
- token generation upon successful signup

Still have some tests to write, but will open a new issue for that.

closes #58 

also closes #17, closes #21, closes #23, closes #25

### CODE REVIEW

File: src/middleware/is-authenticated.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    if (!req.header('x-access-token')) {
+      return res.status(401).send({
+        success: false,
+        message: 'Please login or sign up'
+      });
+    }
+
+    var token = req.header('x-access-token').split(' ')[1];
+
+    return jwt.verify(token, process.env.JWT_SECRET, function(
```
Comment (obay-mardini) [inline src/middleware/is-authenticated.js]: could you put the anonymous function outside verify's parentheses, it's really hard to follow in current situation

**Review by obay-mardini — CHANGES_REQUESTED** [review]

**Review by obay-mardini — APPROVED** [review]

**Comment by obay-mardini** [comment]
LGTM

**Comment by gzeegz** [comment]
Just realized that this also closes #17 #21 #23 #25
